src/distribution/robust_soliton.rs

```diff
@@ -0,0 +1,345 @@
+use crate::distribution::{Soliton, IdealSoliton};
+use crate::statistics::*;
+use crate::{Result, StatsError};
+use rand::distributions::Distribution;
+use rand::Rng;
+use std::f64;
+
+/// Implements the [Robust Soliton]
+/// (https://en.wikipedia.org/wiki/Soliton_distribution)
+/// distribution
+///
+/// # Examples
+///
+/// ```
+/// use statrs::distribution::{Soliton, IdealSoliton};
+/// let n = RobustSoliton::new(10, true, 0.1, 0.1).unwrap();
+/// assert_eq!(n.soliton(5), 0.05879983550709325);
+/// ```
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct RobustSoliton {
+    min: i64,
+    max: i64,
+    spike: Option<i64>,
+    cumulative_probability_table: Vec<f64>,
+    ripple: Ripple,
+    fail_probability: f64,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct RobustSolitonDistribution {
```

Comment:
This struct seems forgotten code after a refactor? It's not used anywhere and has no constructor, but private fields.

Reply:
Yep, missed this. Will remove.